compiler/src/dotty/tools/dotc/core/Annotations.scala

```diff
@@ -117,11 +117,17 @@ object Annotations {
       }
 
     /** Create an annotation where the symbol and the tree are computed lazily. */
-    def deferredSymAndTree(sym: => Symbol, treeFn: Context => Tree)(implicit ctx: Context): Annotation =
+    def deferredSymAndTree(symf: Context => Symbol, treeFn: Context => Tree)(implicit ctx: Context): Annotation =
       new LazyAnnotation {
-        lazy val symf = sym
-
-        override def symbol(implicit ctx: Context): Symbol = symf
+        private[this] var mySym: Symbol = _
+
+        override def symbol(implicit ctx: Context): Symbol = {
+          if (mySym == null) {
```

Comment:
This one looks fishy. Should we not recompute the symbol here maewhen its validity period does not fall in the current run? ImportInfo has a similar problem, but there I think one can argue that it does not matter, as ImportInfos are either global (live forever) or are local to one compilation unit. But for annotations I don't think we can make that argument.